### ISSUE TITLE
Incorrect separate several shapes

### ISSUE THREAD

**Issue by mamadaevv** [PEZ/SketchSeparateShapes#8]
We have 3 overlaping ovals. And we see 7 areas.
![1mdpi](https://user-images.githubusercontent.com/4017693/36036557-3beab2d2-0dcb-11e8-8f13-65eded30f347.png)

I think, after separate we should have 7 shapes:
![correctmdpi](https://user-images.githubusercontent.com/4017693/36036836-f804e410-0dcb-11e8-8520-cf33ade135d4.png)

But in real, we have 5 shapes:
![incorrectmdpi](https://user-images.githubusercontent.com/4017693/36036881-1d613236-0dcc-11e8-898e-8179d2cc402e.png)

With animation (for more clear):
![screencast 2018-02-09 19-35-36](https://user-images.githubusercontent.com/4017693/36038418-b57e83bc-0dd0-11e8-9741-e6832d45c6dc.gif)

**Comment by mamadaevv** [comment]
I tried to play with other options:
![legendmdpi](https://user-images.githubusercontent.com/4017693/36039345-46776558-0dd3-11e8-8151-a530f1b6fda9.png)

![slicemdpi](https://user-images.githubusercontent.com/4017693/36039419-7af8493c-0dd3-11e8-8b62-f1d2e25eec9b.png)

![slicemdpi](https://user-images.githubusercontent.com/4017693/36039427-806fdbdc-0dd3-11e8-9618-33efb6aea2ed.png)

![slicemdpi](https://user-images.githubusercontent.com/4017693/36039433-84f8d6b8-0dd3-11e8-8380-cb2e6fd7eace.png)

**Comment by PEZ** [comment]
I agree it would be desirable with this behaviour. But it is quite clearly stated in the readme what to expect: The bottommost shape will be considered the _base_ and all other shapes will get separated from this shape, and not from each other.

I will keep figuring on how to separate 'em all. And I think your specs here will help a lot with thinking it through. Thanks!

**Comment by PEZ** [comment]
@Mikola35 , can you describe the steps you would take, using only Sketch built in tools, to create the correct results here. After all, this was how we started with this plugin, you describing the manual process.

**Comment by mamadaevv** [comment]
@PEZ I tried it again with three or more shapes but it really difficult. So many corner cases. Now I don't see clear algorithm for this issue. I will try it again later.
